ansible/playbooks/roles/rook/tasks/main.yml

```diff
@@ -0,0 +1,67 @@
+---
+- name: Prepare configuration and upgrade/install Rook Helm chart
+  when: specification.enabled
+  become: true
+  run_once: true
+  block:
+
+    - name: RedHat fix | Create helm's binary symlink
+      file:
+        src: "/usr/local/bin/helm"
+        dest: "/usr/bin/helm"
+        state: link
+      when: ansible_os_family == 'RedHat'
+
+    - name: Download Rook's Chart Files
+      include_role:
+        name: download
+        tasks_from: download_file
+      vars:
+        file_name: "{{ item }}"
+      loop:
+        - "{{ rook_helm_chart_file_name }}"
+        - "{{ rook_helm_cluster_chart_file_name }}"
+
+    - name: Create custom configuration for operator Helm chart file (operator-custom-chart-values.yml)
+      when: specification.operator_chart_values is defined
+      copy:
+        content: "{{ specification.operator_chart_values }}"
+        dest: "{{ download_directory }}/operator-custom-chart-values.yml"
+
+    - name: Create custom configuration for cluster Helm chart file (cluster-custom-chart-values.yml)
+      when: specification.cluster_chart_values is defined
+      copy:
+        content: "{{ specification.cluster_chart_values }}"
+        dest: "{{ download_directory }}/cluster-custom-chart-values.yml"
+
+    - name: Install Rook operator using Helm chart with values from operator-custom-chart-values.yml
+      when: specification.operator_chart_values is defined
+      shell: |
+        helm -n {{ rook_chart_namespace }} upgrade --install \
+          -f {{ download_directory }}/operator-custom-chart-values.yml \
+          {{ rook_helm_chart_name }} \
+          {{ download_directory }}/{{ rook_helm_chart_file_name }} --create-namespace
+
+    - name: Install Rook operator using Helm chart with default values
+      when: not specification.operator_chart_values is defined
+      shell: |
+        helm -n {{ rook_chart_namespace }} upgrade --install \
+          {{ rook_helm_chart_name }} \
+          {{ download_directory }}/{{ rook_helm_chart_file_name }} --create-namespace
+
+    - name: Create Rook cluster with values from cluster-custom-chart-values.yml
+      when: specification.cluster_chart_values is defined
+      shell: |
+        helm -n {{ rook_chart_namespace }} upgrade --install \
+          --set operatorNamespace={{ rook_chart_namespace }} \
+          -f {{ download_directory }}/cluster-custom-chart-values.yml \
+          {{ rook_helm_cluster_chart_name }} \
+          {{ download_directory }}/{{ rook_helm_cluster_chart_file_name }} --create-namespace
+
+    - name: Create Rook cluster with default values
+      when: not specification.cluster_chart_values is defined
+      shell: |
+        helm -n {{ rook_chart_namespace }} upgrade --install \
+          --set operatorNamespace={{ rook_chart_namespace }} \
+          {{ rook_helm_cluster_chart_name }} \
+          {{ download_directory }}/{{ rook_helm_cluster_chart_file_name }} --create-namespace
```

Comment:
Same as above.

Reply:
Same as above. :)

Reply:
I think we could use command module instead of shell here